skyportal/handlers/api/phot_stat.py

```diff
@@ -0,0 +1,195 @@
+import sqlalchemy as sa
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    Obj,
+    Photometry,
+    PhotStat,
+)
+from baselayer.app.access import permissions, auth_or_token
+
+
+class PhotStatHandler(BaseHandler):
+    @auth_or_token
+    def get(self, obj_id=None):
+        """
+        ---
+        description: retrieve the PhotStat associated with the obj_id.
+        tags:
+          - photometry
+        parameters:
+          - in: path
+            name: obj_id
+            required: true
+            schema:
+              type: string
+            description: object ID to get statistics on
+        responses:
+          200:
+            content:
+              application/json:
+                schema: PhotStat
+          400:
+              content:
+                application/json:
+                  schema: Error
+
+        """
+        obj = Obj.get_if_accessible_by(obj_id, self.current_user)
+        if obj is None:
+            return self.error(f'Cannot find source with id "{obj_id}". ')
+
+        with DBSession() as session:
+            stmt = sa.select(PhotStat).where(PhotStat.obj_id == obj_id)
+            phot_stat = session.scalars(stmt).first()
+
+            if phot_stat is None:
+                return self.error(
+                    f'Could not find a PhotStat for object with id "{obj_id}". '
+                )
+
+            stmt = (
+                sa.select(Photometry)
+                .where(Photometry.obj_id == obj_id)
+                .order_by(Photometry.created_at.desc())
+            )
+            last_photometry = session.scalars(stmt).first()
+            if last_photometry:
+                phot_stat.last_phot_add_time = last_photometry.created_at
+            else:
+                phot_stat.last_phot_add_time = None
```

Comment:
Are you also trying to save the update to `phot_stat` here or just update the dictionary and return it? Seems like there are other statistics that could also be updated?

Reply:
The idea is that PhotStats is already updated when the photometry is being inserted (or patched or deleted). In this part of the code I'm just giving the user an opportunity to check that the PhotStat object has been updated close after the last photometry was posted.